chelsea-marco/server.js

```diff
@@ -0,0 +1,194 @@
+      request.body.authorUrl,
+      request.body.category,
+      request.body.publishedOn,
+      request.body.body
+    ]
+  )
+  .then(function() {
+    response.send('insert complete')
+  })
+  .catch(function(err) {
+    console.error(err);
+  });
+});
+
+app.put('/articles/:id', function(request, response) {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This is 3 and 4 and 5 of the diagram again, for the same reasons as the last question. This interacts with .updateRecord(). Because you are making changes to the DB via an SQL query, then getting a confirmation from the DB. It is "U" of CRUD, because it is an update.
+  client.query(
+    `UPDATE articles
+    SET
+      title=$1, author=$2, "authorUrl"=$3, category=$4, "publishedOn"=$5, body=$6
+    WHERE article_id=$7;
+    `,
+    [
+      request.body.title,
+      request.body.author,
+      request.body.authorUrl,
+      request.body.category,
+      request.body.publishedOn,
+      request.body.body,
+      request.params.id
+    ]
+  )
+  .then(function() {
+    response.send('update complete')
+  })
+  .catch(function(err) {
+    console.error(err);
+  });
+});
+
+app.delete('/articles/:id', function(request, response) {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This is part 3, 4, and 5 of the diagram, for the same reasons. It interacts with .deleteRecord(). Deleting from database via query, then sending confirmation. This is "D" of CRUD -- deleting.
+  client.query(
+    `DELETE FROM articles WHERE article_id=$1;`,
+    [request.params.id]
+  )
+  .then(function() {
+    response.send('Delete complete')
+  })
+  .catch(function(err) {
+    console.error(err);
+  });
+});
+
+app.delete('/articles', function(request, response) {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This is 3, 4, and 5 of the diagram and "D" (delete) in crud. It interacts with .truncateTable().
+  client.query(
+    'DELETE FROM articles;'
+  )
+  .then(function() {
+    response.send('Delete complete')
+  })
+  .catch(function(err) {
+    console.error(err);
+  });
+});
+
+// COMMENT: What is this function invocation doing?
+// It is envoking loadDB. If there is no database, it creates one. If there is a database, it calls loadArticles, which inserts a bunch of HackerIpsum as database content for you to work with.
```

Comment:
There will always be a DB, your server will not load otherwise if the DB specified in your con-string is not there (kilovolt or blog_DB). This creates the tables (articles,author) if they don't exist.